source/extensions/filters/network/zookeeper_proxy/zookeeper_decoder.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <cstdint>
+
+#include "envoy/common/platform.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+enum class XidCodes {
+  CONNECT_XID = 0,
+  WATCH_XID = -1,
+  PING_XID = -2,
+  AUTH_XID = -4,
+  SET_WATCHES_XID = -8
+};
+
+enum class OpCodes {
+  CONNECT = 0,
+  CREATE = 1,
+  DELETE = 2,
+  EXISTS = 3,
+  GETDATA = 4,
+  SETDATA = 5,
+  GETACL = 6,
+  SETACL = 7,
+  GETCHILDREN = 8,
+  SYNC = 9,
+  PING = 11,
+  GETCHILDREN2 = 12,
+  CHECK = 13,
+  MULTI = 14,
+  CREATE2 = 15,
+  RECONFIG = 16,
```

Comment:
ZK 3.5 added check/remove watches - I don't remember seeing in the doc - which version of ZK this is supporting? Worth mentioning and possibly adding those types here? (depending on your intent)

...
#define ZOO_RECONFIG_OP 16
#define ZOO_CHECK_WATCHES 17
#define ZOO_REMOVE_WATCHES 18
#define ZOO_CLOSE_OP -11
...

Reply:
@phunt I am targeting ZK > =3.5, which I'll comment in the docs. I'll add support for check/remove watches and I'll also explicitly handle unknown opcodes, so we are covered for the future.

Reply:
Done.